typescript/alert-center/src/frontend.ts

```diff
@@ -0,0 +1,69 @@
+import { ArgOptional, GetApi, HandlerContext, PostApi } from "@dbos-inc/dbos-sdk";
+import path from 'path';
+import { Liquid } from "liquidjs";
+import { RespondUtilities, AlertStatus } from "./utilities";
+import { Respondment, AlertEmployeeInfo } from "./operations";
+
+const engine = new Liquid({
+  root: path.resolve(__dirname, '..', 'public'),
+  extname: ".liquid"
+});
+
+async function render(file: string, ctx?: object): Promise<string> {
+  return await engine.renderFile(file, ctx) as string;
+}
+
+export class Frontend {
+
+  @GetApi('/')
+  static frontend(_ctxt: HandlerContext) {
+    return render("app.html", {});
+  }
+
+  @GetApi('/assignment')
+  static async getAssignment(ctxt: HandlerContext, name: string, @ArgOptional more_time: boolean | undefined) {
+    const userRecWF = await ctxt.startWorkflow(Respondment).userAssignmentWorkflow(name, more_time);
+    const userRec = await ctxt.getEvent<AlertEmployeeInfo>(userRecWF.getWorkflowUUID(), 'rec');
+    return userRec;
+  }
+
+  @GetApi('/alert_history')
+  static async alerts(ctxt: HandlerContext) {
+    const {alerts, employees} = await ctxt.invoke(RespondUtilities).popDashboard();
+    return alerts;
+  }
+  
+  @PostApi('/respond/cancel')
+  static async cancelAssignment(ctxt: HandlerContext, name: string) {
+    await ctxt.invoke(RespondUtilities).employeeAbandonAssignment(name);
+    return Promise.resolve();
+  }
+  
+  @PostApi('/respond/fixed') 
+  static async fixAlert(ctxt: HandlerContext, name: string) {
+    await ctxt.invoke(RespondUtilities).employeeCompleteAssignment(name);
+  }
+  
+  @PostApi('/respond/more_time')
```

Comment:
As mentioned above, I'd suggest removing this endpoint.

Reply:
See prev comment